src/plugins/mer/mervirtualboxmanager.cpp

```diff
@@ -500,6 +518,91 @@ void MerVirtualBoxManager::setVideoMode(const QString &vmName, const QSize &size
     process->runAsynchronously(args);
 }
 
+bool MerVirtualBoxManager::setVdiCapacityMb(MerConnection *connection, int sizeMb)
+{
+    const QString &vmName = connection->virtualMachine();
+    qCDebug(Log::vms) << "VM:" << vmName << "Changing vdi size to" << sizeMb << "MB";
+
+    const VirtualMachineInfo virtualMachineInfo = fetchVirtualMachineInfo(vmName, true);
+    if (sizeMb < virtualMachineInfo.vdiCapacityMb)
+        return false;
+    else if (sizeMb == virtualMachineInfo.vdiCapacityMb)
+        return true;
+
+    QStringList arguments;
+    arguments.append(QLatin1String(MODIFYMEDIUM));
+    arguments.append(virtualMachineInfo.vdiPath);
+    arguments.append(QLatin1String(RESIZE));
+    arguments.append(QString::number(sizeMb));
+
+    QTime timer;
+    timer.start();
+
+    VBoxManageProcess process;
+    if (!process.runSynchronously(arguments)) {
```

Comment:
Yes, however, the virtual machine restarts during this function. It may be don't obvious for user why change VM states. Do it anyway?

Reply:
Asynchronous so that it does not block the event loop. The VM will be locked down using MerConnection::lockDown, so starting from qtc will not be possible. Also the progress dialog will be modal, so the user will not be able to interact with qtc at all. It is just about to not block repainting the UI.

Reply:
Now resize vdi takes time as changing memory and cpu. It take a lot time when restart VM and change partition table inside, but now it was on start VM.

Reply:
OK, hopefully it will be always that fast, even when the VDI is bigger and/or the VM is used for longer time.